packages/relay/src/lib/services/debugService/index.ts

```diff
@@ -0,0 +1,118 @@
+import { Logger } from 'pino';
+import { MirrorNodeClient } from '../../clients';
+import { IDebugService } from './IDebugService';
+import { CacheService } from '../cacheService/cacheService';
+import { predefined } from '../../errors/JsonRpcError';
+import { CommonService } from '../ethService/ethCommonService';
+
+export class DebugService implements IDebugService {
+  /**
+   * The interface through which we interact with the mirror node
+   * @private
+   */
+  private readonly mirrorNodeClient: MirrorNodeClient;
+
+  /**
+   * The logger used for logging all output from this class.
+   * @private
+   */
+  private readonly logger: Logger;
+
+  /**
+   * The LRU cache used for caching items from requests.
+   *
+   * @private
+   */
+  private readonly cacheService: CacheService;
+  private readonly common: CommonService;
+  public readonly debugTraceTransaction = 'debug_traceTransaction';
+
+  constructor(mirrorNodeClient: MirrorNodeClient, logger: Logger, cacheService: CacheService, common: CommonService) {
+    this.mirrorNodeClient = mirrorNodeClient;
+    this.logger = logger;
+    this.cacheService = cacheService;
+    this.common = common;
+  }
+
+  /**
+   * Checks if the Filter API is enabled
+   */
+  static requireDebugAPIEnabled(): void {
+    if (!process.env.DEBUG_API_ENABLED || process.env.DEBUG_API_ENABLED !== 'true') {
+      throw predefined.UNSUPPORTED_METHOD;
+    }
+  }
+
+  async debug_traceTransaction(
+    transactionHash: string,
+    tracer: any,
+    tracerConfig: any,
+    requestIdPrefix?: string
+  ): Promise<any> {
+    this.logger.trace(`${requestIdPrefix} debug_traceTransaction(${transactionHash})`);
+    try {
+      DebugService.requireDebugAPIEnabled();
+      if (tracer.tracer === 'callTracer') {
+        return await this.callTracer(transactionHash, tracerConfig, requestIdPrefix);
+      } else if (tracer.tracer === 'opcodeLogger') {
+        throw Error('opcodeLogger is currently not supported');
+      }
```

Comment:
do we support `prestateTracer` ?

Reply:
it has not been included in the design discussion, so I think no? The plan is only for callTracer and opcodeLogger